applications/nexus/nex2ecl.cpp

```diff
@@ -16,37 +16,148 @@
 */
 
 #include <cctype>
-
-#include <string>
-
+#include <getopt.h>
 #include <iostream>
+#include <string>
 
 #include <ert/util/util.h>
 
 #include "nexus/util.hpp"
 
+namespace {
+
+static const constexpr char *usage =
+R"(nex2ecl [options]... <nexus_plt_file>
+Try 'nex2ecl --help' for more information.)";
+
+static const constexpr char *help =
+R"(Usage:
+  nex2ecl [options]... <nexus_plt_file>
+  nex2ecl -h | --help
+Options:
+  -f --format                Format the output.
+  -h --help                  Show this screen.
+  -o --output <file>|<path>  Write output to files prefixed by <file> or place
+                             output at <path>.
+)";
+
+static struct option long_options[] = {
+    {"format", no_argument,       nullptr, 'f'},
+    {"help",   no_argument,       nullptr, 'h'},
+    {"output", required_argument, nullptr, 'o'}
+};
 
-void create_ecl_sum(char *file_path) {
-    std::string filename(file_path);
-    auto ending = filename.rfind(".plt");
+}
+
+void create_ecl_sum( std::string input_path,
+                     std::string output_path,
+                     bool format_output ) {
+    auto ending = input_path.rfind(".plt");
     if (ending == std::string::npos) {
-        filename += ".plt";
+        input_path += ".plt";
     }
 
-    nex::NexusPlot plt = nex::load(filename);
+    std::string ecl_case_name {};
+    {
+        char *plt_basename;
+        util_alloc_file_components( input_path.c_str(),
+                                    NULL,
+                                    &plt_basename,
+                                    NULL );
+        util_strupr( plt_basename );
+
+        if ( output_path.empty() ) {
+            ecl_case_name = std::string( plt_basename );
+        } else if ( util_is_directory(output_path.c_str()) ) {
+            ecl_case_name = output_path + std::string( plt_basename );
+        } else {
+            char *path = 0, *basename = 0;
+            util_alloc_file_components(output_path.c_str(), &path, &basename, NULL);
+
+            if ( !basename ) {
+                std::cerr << "Failed to parse output path "
+                          << output_path << "." << std::endl;
+                exit(1);
+            }
+
+            if ( path && !util_is_directory(path) ) {
+                std::cerr << path << " does not exists." << std::endl;
+                exit(1);
+            }
+
+            util_strupr( basename );
 
-    char * basename;
-    util_alloc_file_components( file_path , NULL, &basename , NULL);
-    util_strupr( basename );
+            if ( path ) {
+                ecl_case_name += std::string( path );
+                ecl_case_name += UTIL_PATH_SEP_CHAR;
+            }
+            ecl_case_name += std::string( basename );
 
-    ecl_sum_type *ecl_sum = nex::ecl_summary(basename, plt);
+            if ( path ) free( path );
```

Comment:
As long as `path` and `basename`have been ǸULL initialized you *can* call `free( )` unconditionally.